src/lava/lib/dl/netx/blocks/process.py

```diff
@@ -19,23 +19,27 @@ class AbstractBlock(AbstractProcess):
     ----------
     shape : tuple or list
         shape of the block output in (x, y, z) or WHC format.
-    neuron_params : dict
+    neuron_params : dict, optional
         dictionary of neuron parameters. Defaults to None.
-    has_graded_input : dict
-        flag for graded spikes at input. Defaults to False.
+    input_message_bits : int, optional
+        number of message bits in input spike. Defaults to 0 meaning unary
+        spike.
```

Comment:
shouldn't unary spike default to 1 rather than 0?

Reply:
Unary spikes don't use any message bits. The fact that the destination gets a spike means that the spike has a value of 1 and 0 if it does not reach. It is the mode of spiking in Loihi 1 and `num_message_bits=0` in Loihi 2.

When `num_message_bits=1`, a spike can have a payload of 0 which is equivalent to not sending a spike. That is why the correct term IMO is __unary spike__, not binary spike.

Reply:
I see, then we can just say "default 0 because unary spikes have no-payload" or something like that.